src/besselk.jl

```diff
@@ -271,3 +271,52 @@ function besselk_ratio_knu_knup1(v, x::T) where T
     xinv = inv(x)
     return xinv * (v + x + 1/2) + xinv * hn
 end
+
+# originally contribued by @cgeoga https://github.com/cgeoga/BesselK.jl/blob/main/src/besk_ser.jl
+function _besselk_ser(v, x, maxit, tol, modify)
+  T    = promote_type(typeof(x), typeof(v))
+  out  = zero(T)
+  oneT = one(T)
+  twoT = oneT+oneT
+  # precompute a handful of things:
+  xd2      = x/twoT
+  xd22     = xd2*xd2
+  half     = oneT/twoT
+  if modify
+    e2v      = exp2(v)
+    xd2_v    = (x^(2*v))/e2v
+    xd2_nv   = e2v
+  else
+    lxd2     = log(xd2)
+    xd2_v    = exp(v*lxd2)
+    xd2_nv   = exp(-v*lxd2)
+  end
+  gam_v    = gamma(v)
+  gam_nv   = gamma(-v)
```

Comment:
this can use the reflection formula.

Reply:
fixed this (may want to also include a check that v can not be an integer but we check in the top level)

Reply:
If this is an internal function, I would just document it. `isinteger` isn't totally free.

Reply:
Yes it's internal. We dispatch to a different method for integer orders at top level. Will add notes to reflect that